### ISSUE TITLE
Tagged release builds are not publishing and promoting

### ISSUE THREAD

**Issue by travisn** [rook/rook#8378]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The `v1.7.0-beta.0` tag was pushed to github, the build was triggered as expected, but the publish and promote actions were not completed. From the log it seems they are no-ops. See the [full log](https://github.com/rook/rook/runs/3147646826?check_suite_focus=true).

```
+ build/run make -C build/release build BRANCH_NAME=v1.7.0-beta.0 GIT_API_TOKEN=***
==== building the cross container (this could take minutes the first time)
make: Entering directory '/home/rook/go/src/github.com/rook/rook/build/release'
make: Leaving directory '/home/rook/go/src/github.com/rook/rook/build/release'
+ build/run make -C build/release publish BRANCH_NAME=v1.7.0-beta.0 AWS_ACCESS_KEY_ID=*** AWS_SECRET_ACCESS_KEY=*** GIT_API_TOKEN=***
+ git status
+ git diff
HEAD detached at v1.7.0-beta.0
nothing to commit, working tree clean
==== building the cross container (this could take minutes the first time)
make: Entering directory '/home/rook/go/src/github.com/rook/rook/build/release'
make: Leaving directory '/home/rook/go/src/github.com/rook/rook/build/release'
+ build/run make -C build/release promote BRANCH_NAME=v1.7.0-beta.0 CHANNEL=release AWS_ACCESS_KEY_ID=*** AWS_SECRET_ACCESS_KEY=***
==== building the cross container (this could take minutes the first time)
make: Entering directory '/home/rook/go/src/github.com/rook/rook/build/release'
make: Leaving directory '/home/rook/go/src/github.com/rook/rook/build/release'
```

The `publish` function works perfectly fine for pushes to the release branch, and the `publish_and_promote` works fine for the master branch pushes. The script seems triggered correctly for the tag, but something is missing...

For a healthy build log from a push to the release branch, see [here](https://github.com/rook/rook/runs/3147519845?check_suite_focus=true).

**Expected behavior:**
The tagged builds should be publishing and promoting the build.

**How to reproduce it (minimal and precise):**

Push a tag to the repo

**Comment by subhamkrai** [comment]
@travisn @leseb I was looking to action run
one doubt,
here(on the unsuccessful run ) it should branch name https://github.com/rook/rook/runs/3147646826?check_suite_focus=true#step:8:1353 and to me looks like it picked the tag instead of the branch name.
`build/run make -C build/release build BRANCH_NAME=v1.7.0-beta.0 GIT_API_TOKEN=***`


on success full run [here](https://github.com/rook/rook/runs/3147519845?check_suite_focus=true#step:8:1354) it is 
`+ build/run make -C build/release build BRANCH_NAME=release-1.7 TAG_WITH_SUFFIX=true GIT_API_TOKEN=***`
 
is my observation right?

**Comment by subhamkrai** [comment]
> @travisn @leseb I was looking to action run
> one doubt,
> here(on the unsuccessful run ) it should branch name https://github.com/rook/rook/runs/3147646826?check_suite_focus=true#step:8:1353 and to me looks like it picked the tag instead of the branch name.
> `build/run make -C build/release build BRANCH_NAME=v1.7.0-beta.0 GIT_API_TOKEN=***`
> 
> on success full run [here](https://github.com/rook/rook/runs/3147519845?check_suite_focus=true#step:8:1354) it is
> `+ build/run make -C build/release build BRANCH_NAME=release-1.7 TAG_WITH_SUFFIX=true GIT_API_TOKEN=***`
> 
> is my observation right?

reading this 
```
The branch or tag ref that triggered the workflow run. For branches this in the format refs/heads/<branch_name>, and for tags it is refs/tags/<tag_name>
```
from Github docs says `github.ref` will pick `tag_name` for tags

**Comment by travisn** [comment]
We do want to pull the repo based on the tag name, then use that tag of course to tag images. But I see now that the branch name was used in the Jenkins build and the tag was picked up in a separate place. The build really should be done based on the tag in case we need to build on it instead of the head of the release branch, but i'll dig a bit deeper...

**Comment by subhamkrai** [comment]
What I understand, what we need is the name of the branch from which tag was pushed. Let's discuss offline

**Comment by TomHellier** [comment]
Could we get the helm chart released as well? interested in trying out the new rook-ceph-cluster chart before 1.7.0 is finished?

**Comment by travisn** [comment]
> Could we get the helm chart released as well? interested in trying out the new rook-ceph-cluster chart before 1.7.0 is finished?

@TomHellier Yes, i'm investigating getting the new helm chart published with v1.7.0-beta.0. The build is running now and fingers crossed we will have the chart published in the next half hour!

**Comment by travisn** [comment]
I don't see it there, will keep looking for the fix...

**Comment by travisn** [comment]
Ok the charts are published now with the v1.7.0-beta.1 tag!
